cpp/src/barretenberg/dsl/acir_format/block_constraint.cpp

```diff
@@ -0,0 +1,63 @@
+#include "block_constraint.hpp"
+//#include "round.hpp"
+//#include "stdlib/types/types.hpp"
+#include "barretenberg/stdlib/primitives/memory/rom_table.hpp"
+#include "barretenberg/stdlib/primitives/memory/ram_table.hpp"
+// using namespace plonk::stdlib::types;
+
+using namespace proof_system::plonk;
+
+namespace acir_format {
+field_ct poly_to_field_ct(const poly_triple poly, Composer& composer)
+{
+    ASSERT(poly.q_m == 0);
+    ASSERT(poly.q_r == 0);
+    ASSERT(poly.q_o == 0);
+    if (poly.q_l == 0) {
+        return field_ct(poly.q_c);
+    }
+    field_ct x = field_ct::from_witness_index(&composer, poly.a);
+    x.additive_constant = poly.q_c;
+    x.multiplicative_constant = poly.q_l;
+    return x;
+}
+
+void create_block_constraints(Composer& composer, const BlockConstraint constraint)
```

Comment:
Let's add a basic test into the C++, like we do in `aztec_backend` that has block constraints, create's a circuit with a witness and verifies a proof. Similar to all the tests in `acir_format.test.cpp` with: 
```
    auto prover = composer.create_ultra_with_keccak_prover();
    auto proof = prover.construct_proof();
    auto verifier = composer.create_ultra_with_keccak_verifier();
    EXPECT_EQ(verifier.verify_proof(proof), true);
``` 
Going forward we will require any acir format constraints to have a matching test in C++.

Reply:
I have added a unit test, however I am not sure if it is properly added to the test suite?

Reply:
I successfully ran the block constraint test with `cmake --build --preset default --target run_acir_format_tests`